File: frontend/src/types/resume.ts
export interface ResumeInfo {
  filename: string
  parsed: {
    skills: string[]
    experience_years: number | null
    education: string | null
    raw_text: string
  }
  uploaded_at: string
}

export interface ResumeUploadResponse extends ResumeInfo {
  cleared: {
    reports_deleted: number
    messages_deleted: number
  }
}

File: backend/app/schemas/resume.py
from pydantic import BaseModel


class ParsedResume(BaseModel):
    skills: list[str] = []
    experience_years: int | None = None
    education: str | None = None
    raw_text: str = ""


class ClearResult(BaseModel):
    reports_deleted: int = 0
    messages_deleted: int = 0


class ResumeOut(BaseModel):
    id: str
    label: str
    filename: str
    parsed: ParsedResume
    is_default: bool = False
    uploaded_at: str


class ResumeUploadOut(ResumeOut):
    cleared: ClearResult = ClearResult()


class ResumeUpdate(BaseModel):
    label: str | None = None
    is_default: bool | None = None
